XdistroSetup.sh

```diff
@@ -0,0 +1,205 @@
+      logger -i -t "${PROGNAME}" -p kern.crit -s -- "${ERRSTR}"
+   else
+      logger -i -t "${PROGNAME}" -p kern.crit -- "${ERRSTR}"
+   fi
+
+   # Only exit if requested exit is numerical
+   if [[ ${SCRIPTEXIT} =~ ${ISNUM} ]]
+   then
+      exit "${SCRIPTEXIT}"
+   fi
+}
+
+
+# Print out a basic usage message
+function UsageMsg {
+   local SCRIPTEXIT
+   SCRIPTEXIT="${1:-1}"
+
+   (
+      echo "Usage: ${0} [GNU long option] [option] ..."
+      echo "  Options:"
+      printf '\t%-4s%s\n' '-d' 'Distro nickname (e.g., "Rocky")'
+      printf '\t%-4s%s\n' '-h' 'Print this message'
+      printf '\t%-4s%s\n' '-k' 'List of RPM-validation key-files or RPMs'
+      printf '\t%-4s%s\n' '-r' 'List of repository-related RPMs'
+      echo "  GNU long options:"
+      printf '\t%-20s%s\n' '--distro-name' 'See "-d" short-option'
+      printf '\t%-20s%s\n' '--help' 'See "-h" short-option'
+      printf '\t%-20s%s\n' '--repo-rpms' 'See "-r" short-option'
+      printf '\t%-20s%s\n' '--sign-keys' 'See "-k" short-option'
+   )
+   exit "${SCRIPTEXIT}"
+}
+
+# Install the alt-distro's GPG key(s)
+function InstallGpgKeys {
+   local LOOP
+   LOOP=0
+
+   while [[ LOOP -lt ${#PKGSIGNKEYS[*]} ]]
+   do
+      if [[ ${PKGSIGNKEYS[${LOOP}]} == "" ]] 
+      then
+         break
+      elif [[ ${PKGSIGNKEYS[${LOOP}]} == *.rpm ]]
+      then
+         echo yum install -y "${PKGSIGNKEYS}"
+      else
+	     printf "Installing %s to /etc/pki/rpm-gpg... " \
+           "${PKGSIGNKEYS[${LOOP}]}"
+         cd /etc/pki/rpm-gpg || err_exit "Could not chdir"
+         curl -sOkL "${PKGSIGNKEYS[${LOOP}]}" || err_exit "Download failed"
+         echo "Success"
+         cd "${RUNDIR}"
+      fi
+      LOOP=$(( LOOP + 1))
+   done
+}
+
+function StageDistroRpms {
+   local LOOP
+   LOOP=0
+
+   if [[ ! -d ${HOME}/RPM/${DISTRONAME} ]]
+   then
+      printf "Creating %s... " "${HOME}/RPM/${DISTRONAME}"
+      install -dDm 0755 "${HOME}/RPM/${DISTRONAME}" || \
+        err_exit "Failed to create ${HOME}/RPM/${DISTRONAME}"
+      echo "Success"
+   fi
+
+   (
+     cd "${HOME}/RPM/${DISTRONAME}"
+
+     while [[ LOOP -lt ${#REPORPMS[*]} ]]
+     do
+	printf "fetching %s to %s... " "${REPORPMS[${LOOP}]}" \
+	  "${HOME}/RPM/${DISTRONAME}"
+        curl -sOkL "${REPORPMS[${LOOP}]}"
+        echo "Success"
+
+        LOOP=$(( LOOP + 1))
+     done
+   )
+}
+
+
+
+######################
+## Main program-flow
+######################
+OPTIONBUFR=$( getopt \
+  -o d:hk:r: \
+  --long distro-name:,help,repo-rpms:,sign-keys:, \
+  -n "${PROGNAME}" -- "$@")
+
+eval set -- "${OPTIONBUFR}"
+
+###################################
+# Parse contents of ${OPTIONBUFR}
+###################################
+while true
+do
+   case "$1" in
+      -d|--distro-name)
+            case "$2" in
+               "")
+                  err_exit "Error: option required but not specified"
+                  shift 2;
+                  exit 1
+                  ;;
+               *)
+		  DISTRONAME="${2}"
```

Comment:
bad tab characters. two other lines with tabs in this case statement also

Reply:
Ugh... Need to look at my `.vimrc`